std/format/internal/write.d

```diff
@@ -2071,81 +2069,81 @@ template hasToString(T, Char)
 
     static struct A
     {
-        void toString(Writer)(ref Writer w)
+        void toString(Writer)(ref Writer w) const
```

Comment:
Are all these new `const` needed? Worries me that this change again changes the Phobos API for the user... (like https://github.com/dlang/phobos/pull/8249 did)

Reply:
This is not a user-facing API change. These `toString` methods are used only in unit tests. Making them `const` allows them to be called on `inout` struct instances, and makes it possible to test `hasToString!(inout(A), char)` without introducing excessive code duplication (i.e., copy-pasting the entire `unittest` block).

Reply:
Of course these `const`s are not user facing. "this change" = this PR.
What I mean is, that the addition of `const` may have been necessary to deal with something that _does_ trickle down to users. (The changes in #8249 also appear not user facing for example, yet they are). It removes some of the tested type space.

Reply:
Thanks for clarifying. I've restored the test coverage of unqualified `toString` methods.